app/assets/javascripts/discourse/controllers/full-page-search.js.es6

```diff
@@ -90,7 +91,9 @@ export default Ember.Controller.extend({
     this._searchOnSortChange = false;
     if (term) {
       SortOrders.forEach(order => {
-        if (term.indexOf(order.term) > -1){
+        let term_start      = term.indexOf(order.term)
+        let char_after_term = term[term_start + (order.term || '').length]
+        if (term_start > -1 && (!char_after_term || char_after_term.match(/\s/))){
```

Comment:
@eviltrout not sure what does it mean, still don't understand the suggestion you suggested. 

From what I understand from your suggestion, the `term` you meant is the search query?
For `order.term` it already concatenated with the sorting method (`latest`, `likes`, `views, `latest_topic`)

What I discover, the problem is if at least 2 `order.term` share the same prefix `in this case order:latest and order:latest_topic`.

Can you explain more about the concatenation of `order:` and `term`? I may understand it incorrectly.